### DISCUSSION TITLE
[Qiita CLI] Qiita APIからの403 Forbidden エラーメッセージが不適切

### DISCUSSION THREAD

**Discussion by rmatttu** [increments/qiita-discussions#561] [要望・アイデア・不具合]
Qiita CLI v1.3.0
において、Qiita APIから403 Forbiddenが返ってきた場合、下記メッセージが表示されます。
「Qiitaのアクセストークンが正しいかご確認ください」

401 Unauthorized のエラーメッセージと同じになってしまっています。
403なので、認証は成功しているがアクセス権がない、のはずなので、「記事ファイルに不備がないかご確認ください」などが正しいかなと思います。
[該当のコード箇所](https://github.com/increments/qiita-cli/blob/9cc589213976b324442bcc478b80d71eb70002b4/src/lib/error-handler.ts#L43C17-L43C17)

ケースとして、
Qiita CLIを使っていて、すでに投稿済みかつ公開設定している記事ファイルにおいて、誤って、記事の設定（Front Matter）の`private`を`true`へ変更してしまいました。
`npx qiita publish 記事のファイルのベース名` を実行しましたが、すでに公開設定した記事は、限定共有記事にできないためエラーになります。（Qiita API側からは403 Forbiddenが返される。）
しかし、Qiita CLIは「Qiitaのアクセストークンが正しいかご確認ください」と表示するので、認証系のエラーかと思いしばらくハマってしまいました。

以上、よろしくお願いします。

**Comment by tanaka-asami415** [comment]
@rmatttu 
ご意見ありがとうございます！
ご不便をおかけして申し訳ありません。検討いたします🙇

**Comment by diontools** [comment]
タグにスペースを含んでいると同様に失敗します。
エラーメッセージに失敗原因が表示されるといいですね。

**Comment by wataru86** [comment]
@rmatttu @diontools

お待たせしました。
publish コマンド実行時のエラーメッセージを修正し、「記事ファイルに不備がないかご確認ください」というメッセージも表示するようにしました。

現状では API からエラーの原因を取得できないため、詳細なエラーを表示することはできませんが、API の改善と併せて将来的に検討したいと思っています！
